packages/renderer-react/src/index.js

```diff
@@ -13,8 +12,14 @@ const withReact = (Base = HTMLElement) =>
       };
     }
     renderer(root, call) {
+      this._renderRoot = root;
       render(call(), root);
     }
+    disconnectedCallback() {
+      super.disconnectedCallback && super.disconnectedCallback();
+      unmountComponentAtNode(this._renderRoot);
+      this._renderRoot = null;
```

Comment:
kinda got this habit from React where `ref`s are nulled out when unmounting - prob a good idea to prevent any memory leaks. Could also add this line to the Preact renderer if you want?

Reply:
Yeah, that'd be great. I'll merge this to get it in and we can do that as a separate change. Should we test that it's nulled out you think?

Reply:
Sure, can do. It feels almost like testing an implementation detail (`expect(el._renderRoot).toBeNull()`), but perhaps that's not so bad